code/modules/admin/diagnostics.dm

```diff
@@ -1111,6 +1111,48 @@ proc/debug_map_apc_count(delim,zlim)
 			else
 				processed_areas.len = 0
 
+	lightswitches //This overlay is a kitbash of areas and active areas, code might be odd as a result
+		name = "light switches"
+		help = {"Shows amount of light switches for non-space areas.<br>
+		Red is none, green is at least one, white turfs have a light switch located on them.<br>
+		This overlay looks only at physical location and does <b>not</b> account for light switches that have otherarea set up."}
+		var/list/area/processed_areas
+		var/list/turf/switch_turfs = list() //Turfs with a light switch on them, to be coloured separately
+		GetInfo(var/turf/theTurf, var/image/debugoverlay/img)
+			var/area/area = theTurf.loc
+			var/switchcount = 0
+			var/list/lcolor = hex_to_rgb_list(debug_color_of(area))
+			var/color_factor = 4
+
+			if(!(area in processed_areas))
+				for (var/obj/machinery/light_switch/someswitch in area.contents)
```

Comment:
Approved this a bit early (my bad), but after talking over this with some other devs-
`in area.contents` is *really* slow, as it's iterating on everything in that area. I'd suggest using `area.machines` instead, which will be a much lighter list.